homeassistant/components/homematic/entity.py

```diff
@@ -115,10 +116,21 @@ def _hm_event_callback(self, device, caller, attribute, value):
 
         # Is data needed for this instance?
         if attribute in self._data:
-            # Did data change?
-            if self._data[attribute] != value:
-                self._data[attribute] = value
-                has_changed = True
+            if (
+                attribute in self._channel_map
+                and device == self._channel_map[attribute]
+            ):
+                # Did data change?
+                if self._data[attribute] != value:
```

Comment:
I think we can remove this check, because the core handle that better

Reply:
I'm not up to date what the core is capable of nowadays. I'm open for improvements. 👍